src/Orleans/Storage/RelationalManagementExtensions.cs

```diff
@@ -0,0 +1,145 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using Orleans.Runtime.Storage.Relational;
+using System.Data.Common;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Transactions;
+
+
+namespace Orleans.Runtime.Storage.Management
+{
+    /// <summary>
+    /// Contains some relational database management extension methods.
+    /// </summary>
+    public static class RelationalManagementExtensions
+    {
+        /// <summary>
+        /// Creates a transaction scope in which the storage operates.
+        /// </summary>
+        /// <param name="storage">The storage object.</param>
+        /// <returns>Returns a default transaction scope for the given storage.</returns>
+        /// <remarks>Does not set <c>System.Transactions.TransactionScopeAsyncFlowOption.Enabled">TransactionScopeAsyncFlowOption</c>as it is .NET 4.5.1.
+        /// This is required to support transaction scopes in async-await type of flows.</remarks>
+        public static TransactionScope CreateTransactionScope(this IRelationalStorage storage)
+        {
+            //By default transaction scope is set to serializable and a timeout for one minute.
+            //The timeout is regardless of what has been set on the command object itself and
+            //the query would be rolled back in the end. These defaults are more usable and
+            //can be customized per database.            
+            var transactionOptions = new TransactionOptions { IsolationLevel = IsolationLevel.ReadCommitted, Timeout = TransactionManager.MaximumTimeout };
```

Comment:
With regard to async-await this is only useful starting from .NET 4.5.1 which allows for flowing the transaction context between threads (as tasks are likely running in different threads during their lifetime).